azurerm/internal/services/synapse/synapse_role_assignment_resource.go

```
@@ -163,7 +169,10 @@ func resourceArmSynapseRoleAssignmentDelete(d *schema.ResourceData, meta interfa
 		return err
 	}
 
-	client := synapseClient.AccessControlClient(id.Workspace.Name, environment.SynapseEndpointSuffix)
+	client, err := synapseClient.AccessControlClient(id.Workspace.Name, environment.SynapseEndpointSuffix)
+	if err != nil {
+		return nil
```

whilst this may seem odd, this should return the error here